tutorials/ner_pytorch_medical/README.md

```diff
@@ -105,3 +157,68 @@ We've tested 4 configurations which have been saved to the `configs` folder. All
 | spaCy + Transformer | `configs/spacy_config_trf.cfg` |spaCy `ner` model with Transformer `roberta-base` | 75.20 |
 | PyTorch + Transformer | `configs/config_trf.cfg` | PyTorch based `torch_ner` pipeline with Transformer `roberta-base` | **78.07** |
 
+
+## Annotate new data with Prodigy
+
+The i2b2 data comes from medical records with all personal health information already removed. This is information like patient names, medical ID numbers, Social Security numbers, etc. Imagine we want to update our trained model with new medical records. With medical data, it's imperative to retain patient privacy. That being said, we could use the standard `ner` Prodigy workflows to improve our model, however we would have to have a separate process where someone goes through all the records we want to annotate and remove all personal health information present. 
+
+To improve this manual workflow, we've added a new custom Prodigy recipe that will anonymize medical records for annotators as data is loaded into Prodigy. 
+
+The identification of personal health information is handled by the PII Endpoint of the [Azure Text Analytics Entity Recognition API](https://github.com/MicrosoftDocs/azure-docs/blob/master/articles/cognitive-services/text-analytics/includes/create-text-analytics-resource.md) wrapped in a custom spaCy pipeline component here: `scripts/azure/azure_ner_pipe.py`
+
+This recipe takes an opinionated approach to anonymization. Before examples are shown to the annotator in Prodigy, we run the Azure Entity Recognizer to extract PII entities, then we replace tokens in Prodigy with their corresponding `shape` attribute, while retaining the original text and tokens to send to the model update.
+
+
+So for a simple mock note:
+
+```
+Name: Kabir Khan
+Date: 2022-01-01
+SSN: 444-44-4444
+
+Information: Patient scheduled for surgery to repair tear in left ACL.
+```
+
+The anonymized text shown to the annotator would look like:
+
+```
+Name: Xxxxx Xxxx
+Date: dddd-dd-dd
+SSN: ddd-dd-dddd
+
+Information: Patient scheduled for surgery to repair tear in left ACL.
+```
+
+This retains all the structure and character offsets into the original note during annotation. Once the annotator accepts the example and the internal Prodigy `update` method is called to update our model in the loop, we send the original text instead of the anonymized text.
```

Comment:
I'm afraid the character offsets aren't necessarily retained, because the `shape` of a sequence of similar characters cuts off at 4. So `Kabir` is `Xxxx` but `Kabirkhan` is  also `Xxxxx`. Does that impact how the Prodigy offsets are mapped back onto the original text?

Reply:
Wait really? I had no idea. Can you share some insight on why that is? 

I can write a quick util function to extend this I guess in place of using `shape` directly.

Reply:
You can find the current implementation here: https://github.com/explosion/spaCy/blob/master/spacy/lang/lex_attrs.py#L115

I think it makes sense, when using this attribute as feature. Whether something is a capital + 10 lower-case letters, or 15, makes little difference when trying to determine what kind of word it is.

Reply:
Ahh makes sense, interesting! Glad I know this now